core/field_colour.js

```diff
@@ -138,13 +138,10 @@ Blockly.FieldColour.prototype.DROPDOWN_BACKGROUND_COLOUR = 'white';
  * @package
  */
 Blockly.FieldColour.prototype.initView = function() {
-  Blockly.FieldColour.superClass_.initView.call(this);
```

Comment:
The superclass `initView` function also sets `display: none` if the field is not visible.  That needs to happen for this field as well.

Reply:
Fixed by #2500. It moves that check up to the init method. (Sorry, I probably shouldn't have put these up all at once. I was just really bored over memorial day)